notebook/notebookapp.py

```diff
@@ -300,7 +306,7 @@ def init_settings(self, jupyter_app, kernel_manager, contents_manager,
             allow_password_change=jupyter_app.allow_password_change,
             server_root_dir=root_dir,
             jinja2_env=env,
-            terminals_available=False,  # Set later if terminals are available
+            terminals_available=terminals_available,
```

Comment:
There are three related things, which I think might be getting a bit confused here:

a. The `terminals_enabled` config option, which is True by default.
b. Whether terminado can be imported
c. Whether we are exposing terminals (i.e. enabled *and* importable)

Currently (before this PR), we use `terminals_enabled` for a and `terminals_available` for c (i.e. available to the user). In this PR,`terminals_available` refers to b, and `terminals_enabled` to c.

Reply:
Agreed - see latest commit.